addons/web/models/ir_http.py

```diff
@@ -137,21 +137,38 @@ def _get_content_common(self, xmlid=None, model='ir.attachment', res_id=None, fi
             xmlid=xmlid, model=model, id=res_id, field=field, unique=unique, filename=filename,
             filename_field=filename_field, download=download, mimetype=mimetype, access_token=access_token
         )
+        placeholder_content = None
         if status != 200:
-            return self._response_by_status(status, headers, content)
-        else:
-            headers.append(('Content-Length', len(content)))
-            response = request.make_response(content, headers)
+            if filename and filename.endswith(".mp4") and not content:
+                # Show placeholder video if video should've 404'd
+                placeholder_content = self._video_placeholder()
+                headers.append(('Content-Type', "video/mp4"))
+            else:
+                return self._response_by_status(status, headers, content)
+
+        content = placeholder_content or (content if not isinstance(content, tuple) else "")
+        headers.append(('Content-Length', len(content)))
+
+        # Setting CSP required in order to play same origin videos as BG videos.
+        # It presents no additional security risk when set to 'self', for more info
+        # check https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy/media-src.
+        try:
+            previous_csp = headers.index(('Content-Security-Policy', "default-src 'none'"))
+            headers[previous_csp] = ('Content-Security-Policy', "media-src 'self'")
```

Comment:
So if there is an existing CSP you just... drop it? And everything breaks if said CSP is not specifically `"default-src 'none'"`?

Reply:
> The idea behind this is the following: If we find the "default-src 'none' header, replace it with media-src self

That replacement is the issue, it's not correct because it changes the semantics of the resource, it completely defeats the purpose of having `default-src 'none'`. Even more so because as far as I can tell this replacement gets applied to *every resource being accessed through the contents endpoint*, not just videos (though I would not call this behaviour correct even for just videos).

> It shouldn't break anything in any case, but perhaps you have a better suggestion for how to write this?

The additional policy should be merged with the existing one, if there is one already. And the finding should be based entirely on the header name, as otherwise if the existing CSP header's value changes' this code will add a second CSP header, the exact thing you're apparently trying to avoid.